diff --git a/Common/GPU/thin3d.cpp b/Common/GPU/thin3d.cpp
index ca9d88f..f4a9f58 100644
--- a/Common/GPU/thin3d.cpp
+++ b/Common/GPU/thin3d.cpp
@@ -170,6 +170,6 @@ static const std::vector<ShaderSource> fsTexCol = {
 	{ShaderLanguage::GLSL_1xx,
 	"#ifdef GL_ES\n"
-	"precision lowp float;\n"
+	"precision mediump float;\n"
 	"#endif\n"
 	"#if __VERSION__ >= 130\n"
 	"#define varying in\n"
@@ -208,6 +208,6 @@ static const std::vector<ShaderSource> fsTexColRBSwizzle = {
 	{GLSL_1xx,
 	"#ifdef GL_ES\n"
-	"precision lowp float;\n"
+	"precision mediump float;\n"
 	"#endif\n"
 	"#if __VERSION__ >= 130\n"
 	"#define varying in\n"
@@ -246,6 +246,6 @@ static const std::vector<ShaderSource> fsCol = {
 	{ GLSL_1xx,
 	"#ifdef GL_ES\n"
-	"precision lowp float;\n"
+	"precision mediump float;\n"
 	"#endif\n"
 	"#if __VERSION__ >= 130\n"
 	"#define varying in\n"
